spec/rubocop/cop/lint/redundant_with_object_spec.rb

```diff
@@ -74,4 +74,17 @@
   it 'an object is used as a block argument' do
     expect_no_offenses('ary.each_with_object([]) { |v, o| v; o }')
   end
+
+  context 'when missing argument to each_with_object' do
+    it 'does not choke' do
+      expect_no_offenses('ary.each_with_object { |v, o| v; o }')
+    end
+
+    it 'still detects' do
+      expect_offense(<<-RUBY.strip_indent)
+        ary.each_with_object { |v| v }
+            ^^^^^^^^^^^^^^^^ Use `each` instead of `each_with_object`.
```

Comment:
This cop cannot know the behavior of `each_with_object` method defined by users.
IMO, this cop for `Enumerable#each_with_object` seems better not to register a offense to the user defined method.
See also https://github.com/rubocop-hq/rubocop/pull/6042#issuecomment-399685514.